src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -3304,7 +3304,9 @@ long QgsPostgresProvider::featureCount() const
   // use estimated metadata even when there is a where clause,
   // although we get an incorrect feature count for the subset
   // - but make huge dataset usable.
-  if ( !mIsQuery && mUseEstimatedMetadata )
+  // don't use it on view because it will always return 0 (view has no rows)
+  const QgsPostgresProvider::Relkind type = relkind();
+  if ( !mIsQuery && mUseEstimatedMetadata && type != Relkind::View )
```

Comment:
Please try using the EXPLAIN method to estimate returned rows for queries and views.